action.yml

```diff
@@ -9,6 +9,14 @@ inputs:
     description: 'Development branch pattern'
     required: true
     default: '^(dev|develop|development)$'
+  semestry_staging_branch_pattern:
+    description: 'Semestry staging branch pattern'
+    required: true
+    default: '^([0-9]+\.[0-9]+-staging|[0-9]+\.[0-9]+\.[0-9]+-staging)$'
+  semestry_testing_branch_pattern:
```

Comment:
You could combine the two expressions into one with the third number in parenthesis `(\.[0-9]+)?`.  Or even take the second and third number in one expression. This is a matter of personal preference I think.

Reply:
Yes I agree, the branch regex could be a lot more concise. I have updated this now